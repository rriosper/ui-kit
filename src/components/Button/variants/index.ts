export { default as shape } from './shape'
export { default as size } from './size'
