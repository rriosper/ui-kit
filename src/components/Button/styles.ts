import { createCleanComponent } from '#utils'
import styled, { css } from '@xstyled/styled-components'
import { backgroundColor, th } from '@xstyled/system'
import { Button } from 'reakit'

import { shape as shapeVariant, size as sizeVariant } from './variants'

// Disabled styles
const disabledStyle = ({ disabled }) => disabled && 'opacity: 0.5;'

const CleanButton = createCleanComponent(Button, ['size', 'backgroundColor'])

const getColor = (props) => {
  console.log(th('colors.primary'))
  return css`
  background-color: primary.D1;
`
}

export const SButton = styled(CleanButton)`
  padding: xs;
  transition: default;
  opacity: 0.8;
  height: fit-content;
  width: fit-content;
  ${backgroundColor};

  outline: none;
  border: none;
  display: flex;
  cursor: pointer;
  justify-content: center;
  align-items: center;
  user-select: none;
  color: white;
  user-select: none;


  ${shapeVariant}
  ${disabledStyle}
  ${sizeVariant()}

  :active {
    ${getColor}
  }

  :hover {
    box-shadow: default;
    opacity: 1;
  }
`
