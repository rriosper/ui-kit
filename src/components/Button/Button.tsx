import React from 'react'

import { DEFAULT_BUTTON_STYLES } from '#constants'
import { SButton } from './styles'

interface ButtonProps {
  className?: string;
  size?: string;
  type?: string;
  backgroundColor?: string;
  shape?: string;
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  fill?: string;
  children?: React.ReactNode | React.ReactNode[];
}

/**
 *
 *  Regular button using the logic of Reakit button
 *  {@link https://reakit.io/docs/button/}
 */

class Button extends React.PureComponent<ButtonProps> {
  render () {
    const {
      shape,
      backgroundColor = DEFAULT_BUTTON_STYLES.backgroundColor,
      disabled,
      children,
      className,
      ...props
    } = this.props
    return (
      <SButton
        className={className}
        backgroundColor={backgroundColor}
        shape={shape}
        disabled={disabled}
        {...props}
      >
        {children}
      </SButton>
    )
  }
}

export default Button
