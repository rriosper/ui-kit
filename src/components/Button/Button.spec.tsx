import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
  test('Handlers', () => {
    const onClick = jest.fn()

    const { getByText, asFragment } = render(
      <Button onClick={onClick}>My button!</Button>
    )

    expect(onClick).toHaveBeenCalledTimes(0)
    fireEvent.click(getByText('My button!'))
    expect(onClick).toHaveBeenCalledTimes(1)

    // TODO: Replace by storyshots
    expect(asFragment()).toMatchSnapshot()
  })
})
