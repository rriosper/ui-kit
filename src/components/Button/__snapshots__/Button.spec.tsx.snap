// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Handlers 1`] = `
<DocumentFragment>
  .c0 {
  padding: xs;
  -webkit-transition: default;
  transition: default;
  opacity: 0.8;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: primary;
  outline: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: default;
  min-height: lg;
  min-width: lg;
  font-size: md;
}

.c0 svg {
  width: sm;
  height: sm;
}

.c0:active {
  background-color: primary.D1;
}

.c0:hover {
  box-shadow: default;
  opacity: 1;
}

<button
    class="c0"
    type="button"
  >
    My button!
  </button>
</DocumentFragment>
`;
